srml/contracts/src/rent.rs

```diff
@@ -95,14 +95,26 @@ fn try_evict_or_and_pay_rent<T: Trait>(
 	// The minimal amount of funds required for a contract not to be evicted.
 	let subsistence_threshold = T::Currency::minimum_balance() + <Module<T>>::tombstone_deposit();
 
+	if balance < subsistence_threshold {
+		// The contract cannot afford to leave a tombstone, so remove the contract info altogether.
+		<ContractInfoOf<T>>::remove(account);
+		runtime_io::kill_child_storage(&contract.trie_id);
+		return RentOutcome::Evicted;
+	}
```

Comment:
this changes the current behavior an account with balance such as `existential_deposit < balance < subsistence_threshold` will not gets removed, on the other hand previously such an account would have his money set to 0 (and its contract and stuff would have been removed on OnFreeBalanceZero)

Reply:
Yes, this is true. I think the new behavior makes more sense (the account balance is above existential deposit, why delete?), but perhaps warrants discussion.

Reply:
Yes I understand your point but at the same time a contract cannot be accessed by anything else than his contract, if its contractInfo gets deleted then then the account can't do anything as nobody gots its private key. I feel like it is dust then. but indeed it is more logical

Reply:
You can redeploy the contract with the exact same sender, code, and constructor data to the same address with the balance preserved.

Reply:
oh I never thought about that, true, yes ok. then your implementation is better.

as a note I never thought about that but this property allows a contract to go into a previous state, hmm interesting